openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -774,8 +774,7 @@ Content-Length: 0
 
 The Credential Issuer provides a nonce value in the HTTP response with a 2xx status code and the following parameters included as top-level members in the message body of the HTTP response using the application/json media type:
 
-* `c_nonce`: REQUIRED. String containing a nonce to be used when creating a proof of possession of the key proof (see (#credential-request)).
-* `c_nonce_expires_in`: OPTIONAL. Number denoting the lifetime in seconds of the `c_nonce`. This value serves only as a hint to the Client, indicating how long the Credential Issuer is likely to accept the `c_nonce` as valid.
+* `c_nonce`: REQUIRED. String containing a nonce to be used when creating a proof of possession of the key proof (see (#credential-request)). This value MUST be unpredictable and unique for every response returned from the nonce endpoint.
```

Comment:
We had something like the following in OIDC core:
```suggestion
* `c_nonce`: REQUIRED. String containing a nonce to be used when creating a proof of possession of the key proof (see (#credential-request)). Sufficient entropy MUST be present in the `c_nonce` values used for its lifetime to prevent attackers from guessing values.
```

Reply:
Does it not matter in the context of the entire OID4VCI flow?

Reply:
In OpenID Connect the nonce is created by and send by the client to the AS, and when it gets the id_token back the client checks if it contains the nonce it sent. If an attacker obtains a nonce (and state) they could potentially start a separate flow using the same nonce/state and inject the resulting id_token back into the other flow, or things like that. (Because these requests are being sent through the browser nonce/state could potentially leak or the user phished into starting a new flow with a nonce associated with the attackers session.)

In OpenID4VCI at the credential endpoint the nonce is being fetched from the credential issuer and then included in the proof that the client is in possession of the key in the key proof - the nonce here is all about freshness and showing that it's not an old key proof. The nonces aren't actually bound to the client or session (the nonce endpoint doesn't require a client id or access token).

It probably needs sufficient entropy that it's infeasible for an attacker to pre-generate proofs for all possible values.

Reply:
I agree the applicability of the nonce is different here when compared with OIDC, I'm comfortable with unpredictable being sufficient all be it a little subjective.